### PR TITLE
Add default ServiceMonitor if the CRD is available

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now deploy a default ServiceMonitor object for the operator if the
CRD is available within the cluster. This allows automatic metrics
discovery via the prometheus operator.
#### Which issue(s) this PR fixes:

Refers to #276 
#### Does this PR have test?

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added automatic ServiceMonitor deployment if the CRD is available within the cluster.
```

### CODE REVIEW

File: internal/pkg/manager/spod/spod_controller.go [hunk 1]
```diff
@@ -275,6 +280,22 @@ func (r *ReconcileSPOd) handleCreate(
...
 	}
 
+	hasServiceMonitorCRD, err := r.hasServiceMonitorCRD(ctx)
+	if err != nil {
+		return errors.Wrap(err, "check for ServiceMonitor CRD")
+	}
+
+	if hasServiceMonitorCRD {
+		r.log.Info("Deploying operator service monitor")
+		if err := r.client.Create(
```
Comment (JAORMX) [inline internal/pkg/manager/spod/spod_controller.go:290]: you could just do the call to `Create` and if the error indicates that the object is not registered or is on an unknown type, then you could just ignore it.

e.g.

```
if runtime.IsNotRegisteredError(err) || meta.IsNoMatchError(err) {
```

where the functions come from:

```
	"k8s.io/apimachinery/pkg/runtime"
	"k8s.io/apimachinery/pkg/api/meta"
```

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: internal/pkg/manager/spod/spod_controller.go (hunk 1 above)
Comment (saschagrunert) [reply internal/pkg/manager/spod/spod_controller.go:290]: Ohhh, that’s a good workaround. I’ll change it.

**Pull request merged**